sklearn/datasets/_samples_generator.py

```diff
@@ -306,14 +328,54 @@ def make_classification(
         scale = 1 + 100 * generator.uniform(size=n_features)
     X *= scale
 
+    indices = np.arange(n_features)
     if shuffle:
         # Randomly permute samples
         X, y = util_shuffle(X, y, random_state=generator)
 
         # Randomly permute features
-        indices = np.arange(n_features)
         generator.shuffle(indices)
-        X[:, :] = X[:, indices]
+
+    X[:, :] = X[:, indices]
+
+    # feat_desc describes features in X
+    feat_desc = ["US"] * n_features
+    for i, index in enumerate(indices):
+        if index < n_informative:
+            feat_desc[i] = "IN"
+        elif index >= n_informative and index < n_informative + n_redundant:
+            feat_desc[i] = "RD"
+        elif index >= n and index < n + n_repeated:
+            feat_desc[i] = "RP"
+
+    if not return_X_y:
+        parameters = {
+            "n_samples": n_samples,
+            "n_features": n_features,
+            "n_informative": n_informative,
+            "n_redundant": n_redundant,
+            "n_repeated": n_repeated,
+            "n_classes": n_classes,
+            "n_clusters_per_class": n_clusters_per_class,
+            "weights": weights,
+            "flip_y": flip_y,
+            "class_sep": class_sep,
+            "hypercube": hypercube,
+            "shift": shift,
+            "scale": scale,
+            "shuffle": shuffle,
+            "random_state": random_state,
+        }
+
+        bunch = Bunch(
+            DESCR=make_classification.__doc__,
+            parameters=parameters,
+            feature_info=feat_desc,
+            X=X,
+            y=y,
+        )
+
+        return bunch
 
     return X, y
```

Comment:
you can move this before the previous `if` statement with

```py
if return_X_y:
    return X, y
```

and avoid indentation on the previous block.